x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -508,32 +506,52 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     type: string | string[],
     options: SavedObjectsOpenPointInTimeOptions
   ) {
-    try {
-      const args = { type, options };
-      await this.legacyEnsureAuthorized(type, 'open_point_in_time', options?.namespace, {
+    const args = { type, options };
+    const { status, typeMap } = await this.legacyEnsureAuthorized(
+      type,
+      'open_point_in_time',
+      options?.namespaces,
+      {
         args,
         // Partial authorization is acceptable in this case because this method is only designed
         // to be used with `find`, which already allows for partial authorization.
         requireFullAuthorization: false,
-      });
-    } catch (error) {
+      }
+    );
+
+    if (status === 'unauthorized') {
       this.auditLogger.log(
         savedObjectEvent({
           action: SavedObjectAction.OPEN_POINT_IN_TIME,
-          error,
+          error: new Error(status),
         })
       );
-      throw error;
+      throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));
     }
 
+    const typeToNamespacesMap = Array.from(typeMap).reduce<Map<string, string[] | undefined>>(
+      (acc, [currentType, { authorizedSpaces, isGloballyAuthorized }]) =>
+        isGloballyAuthorized
+          ? acc.set(currentType, options.namespaces)
+          : acc.set(currentType, authorizedSpaces),
+      new Map()
+    );
+
     this.auditLogger.log(
       savedObjectEvent({
         action: SavedObjectAction.OPEN_POINT_IN_TIME,
         outcome: 'unknown',
       })
     );
 
-    return await this.baseClient.openPointInTimeForType(type, options);
+    return await this.baseClient.openPointInTimeForType(
+      status === 'partially_authorized' ? '' : type, // if the user is partially authorized, type must be an empty string to use typeToNamespacesMap instead
```

Comment:
For `find`, `type` is an option. Here, it's a specific call parameter. Top mimic `find`'s `typeToNamespacesMap` usage, I need to pass an empty string. This is ugly, but probably not really more than the same workaround currently in use for `find`, so...

Reply:
See my comment in https://github.com/elastic/kibana/pull/109062#pullrequestreview-734416033, I don't think this change is needed.